### PR TITLE
Fixing partial generic params for specialized generic types

### LINKED ISSUE #1454: Filling generic arguments partially creates incorrect class
Some fallout of https://github.com/microsoft/python-language-server/pull/1404

```python
from typing import Dict, Generic, TypeVar

T = TypeVar('T')
class A(Genetic[T], Dict[T, int]) : ...
```

`A` actually have base of `Dict[T]` rather than proper dictionary with 2 parameters. `Dict[T]` cannot be re-created b/c number of arguments in it is wrong.

```cs
        [TestMethod, Priority(0)]
        public async Task DictPartialParams() {
            const string code = @"
from typing import Dict, Generic, TypeVar

T = TypeVar('T')
class A(Genetic[T], Dict[T, int]) : ...
";

            var analysis = await GetAnalysisAsync(code);
            analysis.Should().HaveClass("A")
                .Which.Should().HaveBase("Dict[T, int]");
        }
```

fails. @CTrando fyi

### PR DESCRIPTION
**Pull request by CTrando**
Fixes #1454

### CODE REVIEW

File: src/Analysis/Ast/Test/FluentAssertions/MemberAssertions.cs [hunk 1]
```diff
@@ -61,20 +61,20 @@ internal class MemberAssertions : ReferenceTypeAssertions<IMember, MemberAsserti
             return new AndWhichConstraint<MemberAssertions, IMember>(this, Subject);
         }
 
-        public AndWhichConstraint<MemberAssertions, IPythonClassType> HaveBase(string name, string because = "", params object[] reasonArgs) {
+        public AndWhichConstraint<MemberAssertions, IPythonType> HaveBase(string name, string because = "", params object[] reasonArgs) {
```
Comment (MikhailArkhipov) [inline src/Analysis/Ast/Test/FluentAssertions/MemberAssertions.cs]: If you want, you can borrow https://github.com/MikhailArkhipov/python-language-server/blob/functools/src/Analysis/Ast/Impl/Specializations/Typing/Types/GenericClassBase.cs which is former `GenericTypeParameter` and is actually a class - as it is supposed to be.

File: src/Analysis/Ast/Test/GenericsTests.cs [hunk 2]
```diff
@@ -1091,6 +1090,37 @@ class Simple(unittest.TestCase):
             analysis.Should().HaveVariable("x").Which.Should().HaveType("TypeError");
         }
 
+        [TestMethod, Priority(0)]
+        public async Task SpecializedGenericPartialParams() {
+            const string code = @"
+from typing import Dict, List, Tuple, Iterator, KeysView, ValuesView, ItemsViews, Sequence, Generic, TypeVar
```
Comment (MikhailArkhipov) [inline src/Analysis/Ast/Test/GenericsTests.cs]: Does the test provided in the issue description pass?

Comment (CTrando) [reply src/Analysis/Ast/Test/GenericsTests.cs]: Yes it does.

File: src/Analysis/Ast/Test/FluentAssertions/MemberAssertions.cs (hunk 1 above)
Comment (CTrando) [reply src/Analysis/Ast/Test/FluentAssertions/MemberAssertions.cs]: If we just want to check if a base is inside the Bases list, then we should just need to pass in the name of the base. By having it as PythonClassType, we can have cases with typing or builtins where the type is a PythonType but we look for PythonClassType. Unless we change Bases to a list of IPythonClassTypes, then I think this is clearer.

File: src/Analysis/Ast/Test/GenericsTests.cs (hunk 2 above)
Comment (CTrando) [reply src/Analysis/Ast/Test/GenericsTests.cs]: Your test is included in this one.

**Pull request merged**